valkyrie/lib/valkyrie/persistence/postgres/orm/resource.rb

```diff
@@ -2,15 +2,21 @@
 module Valkyrie::Persistence::Postgres
   module ORM
     class Resource < ActiveRecord::Base
+      after_save :assign_secondary_identifier, if: ->(obj) { obj.secondary_identifier.blank? }
```

Comment:
Are we forcing a copy? Would `assign_secondary_identifier` be the extension point for different identifier schemes? If not, are we expecting that each place resources are created to assign a secondary ID "if required"? Or do we need an event/callback/collaborator for a system-wide strategy/policy?